pkg/lib/probe/probe.go

```diff
@@ -62,13 +77,72 @@ func NewDefaultProbe(target string, logger *zap.SugaredLogger) *Probe {
 					Host: targetURL.Hostname(),
 				},
 			},
-			TimeoutSeconds: _defaultTimeoutSeconds,
+			InitialDelaySeconds: _initialDelaySeconds,
+			TimeoutSeconds:      _defaultTimeoutSeconds,
+			PeriodSeconds:       _defaultPeriodSeconds,
+			SuccessThreshold:    _defaultSuccessThreshold,
+			FailureThreshold:    _defaultFailureThreshold,
 		},
 		logger: logger,
 	}
 }
 
-func (p *Probe) ProbeContainer() bool {
+func (p *Probe) StartProbing() chan bool {
+	stop := make(chan bool)
+
+	ticker := time.NewTicker(time.Duration(p.PeriodSeconds) * time.Second)
+	time.AfterFunc(time.Duration(p.InitialDelaySeconds)*time.Second, func() {
+		successCount := int32(0)
+		failureCount := int32(0)
+
+		for {
+			select {
+			case <-stop:
+				return
+			case <-ticker.C:
+				healthy := p.probeContainer()
+				if healthy {
+					successCount++
+					failureCount = 0
+				} else {
+					failureCount++
+					successCount = 0
+				}
+
+				if successCount >= p.SuccessThreshold {
+					p.healthy = true
+				} else if failureCount >= p.FailureThreshold {
+					p.healthy = false
+				}
+				p.hasRunOnce = true
+			}
+		}
+	})
+
+	return stop
+}
+
+func (p *Probe) IsHealthy() bool {
+	return p.healthy
+}
```

Comment:
this will have race conditions, needs to be thread safe for realtime. The realtime probe should stay as it was